src/main/java/com/synopsys/integration/alert/web/security/authentication/ldap/LdapManager.java

```diff
@@ -38,64 +42,80 @@
-        final LdapContextSource ldapContextSource = new LdapContextSource();
-
         try {
-            if (StringUtils.isNotBlank(configuration.getServer())) {
-                ldapContextSource.setUrl(configuration.getServer());
-                ldapContextSource.setUserDn(configuration.getManagerDN());
-                ldapContextSource.setPassword(configuration.getManagerPassword());
-                ldapContextSource.setReferral(configuration.getLdapReferral());
-                ldapContextSource.setAuthenticationStrategy(createAuthenticationStrategy(configuration));
+            if (isLdapEnabled()) {
+                final ConfigurationModel configuration = getCurrentConfiguration();
+                final LdapContextSource ldapContextSource = new LdapContextSource();
+
+                final String ldapServer = getFieldValueOrEmpty(configuration, SettingsDescriptor.KEY_LDAP_SERVER);
+                final String managerDN = getFieldValueOrEmpty(configuration, SettingsDescriptor.KEY_LDAP_MANAGER_DN);
+                final String managerPassword = getFieldValueOrEmpty(configuration, SettingsDescriptor.KEY_LDAP_MANAGER_PASSWORD);
+                final String ldapReferral = getFieldValueOrEmpty(configuration, SettingsDescriptor.KEY_LDAP_REFERRAL);
+                if (StringUtils.isNotBlank(ldapServer)) {
+                    ldapContextSource.setUrl(ldapServer);
+                    ldapContextSource.setUserDn(managerDN);
+                    ldapContextSource.setPassword(managerPassword);
+                    ldapContextSource.setReferral(ldapReferral);
+                    ldapContextSource.setAuthenticationStrategy(createAuthenticationStrategy(configuration));
+                }
+                contextSource = ldapContextSource;
+                contextSource.afterPropertiesSet();
+                updateAuthenticationProvider(configuration);
             }
-            contextSource = ldapContextSource;
-            contextSource.afterPropertiesSet();
-            updateAuthenticationProvider();
-        } catch (final IllegalArgumentException ex) {
+        } catch (final IllegalArgumentException | AlertDatabaseConstraintException ex) {
             throw new AlertLDAPConfigurationException("Error creating LDAP Context Source", ex);
         }
     }
 
-    private DirContextAuthenticationStrategy createAuthenticationStrategy(final LdapProperties configuration) {
-        final String authenticationType = configuration.getAuthenticationType();
+    private String getFieldValueOrEmpty(final ConfigurationModel configurationModel, final String fieldKey) {
+        return configurationModel.getField(fieldKey).flatMap(field -> field.getFieldValue()).orElse("");
+    }
+
+    private DirContextAuthenticationStrategy createAuthenticationStrategy(final ConfigurationModel configuration) {
+        final String authenticationType = getFieldValueOrEmpty(configuration, SettingsDescriptor.KEY_LDAP_AUTHENTICATION_TYPE);
```

Comment:
A couple time we're using getFieldValueOrEmpty and then checking if the variable is blank later on. Since we would normally get an optional (And no blank value should be stored in the values table), should we use the optional to check for isPresent instead?